corehq/apps/couch_sql_migration/tests/test_migration.py

```diff
@@ -579,6 +579,23 @@ def test_dry_run(self):
         with self.assertRaises(CommandError):
             call_command('migrate_domain_from_couch_to_sql', self.domain_name, COMMIT=True, no_input=True)
         self.assertFalse(Domain.get_by_name(self.domain_name).use_sql_backend)
+
+        xml = """<?xml version="1.0" ?>
+        <n0:registration xmlns:n0="http://openrosa.org/user/registration">
+            <username>W4</username>
+            <password>2</password>
+            <uuid>P8DU7OLHVLZXU21JR10H3W8J2</uuid>
+            <date>2013-11-19</date>
+            <registering_phone_id>8H1N48EFPF6PA4UOO8YGZ2KFZ</registering_phone_id>
+            <user_data>
+                <data key="user_type">standard</data>
+             </user_data>
+        </n0:registration>
+        """
+        submit_form_locally(xml, self.domain_name)
```

Comment:
I'm not totally sure I understand how this is working. Doesn't `call_command` block until it has completed? So won't this always pass since the dry run has finished once you get to line `578`? Or is there some magic I don't know about that's happening here?

Reply:
Ah yeah, the thing you're missing is that the migration is run in stages, each by a separate run of the management command with a different option. Even if the active run has finished, it won't be totally done until it's called again with `COMMIT` or `blow-away`

Reply:
got it, thanks.